### PR TITLE
[tibber] Query for subscription url every reconnect attempt

### PR DESCRIPTION
**Pull request by kjoglum**
Enhancement to allow binding to check for potential change in subscription URL at every reconnect attempt, not just at initial startup.

### CODE REVIEW

File: bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java [hunk 1]
```diff
@@ -297,7 +297,31 @@ public void updateRequest() throws IOException {
...
+            if (dataObject != null) {
+                JsonObject viewerObject = dataObject.getAsJsonObject("viewer");
+                if (viewerObject != null) {
+                    JsonElement subscriptionElement = viewerObject.get("websocketSubscriptionUrl");
+                    if (subscriptionElement != null) {
+                        subscriptionURL = subscriptionElement.toString().replaceAll("^\"|\"$", "");
+                    }
+                }
+            }
+            String url = subscriptionURL;
```
Comment (jlaur) [inline bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java:316]: This code is duplicated from line 150-164. Could it be extracted to a private method?

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks for the enhancement!

File: bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java (hunk 1 above)
Comment (kjoglum) [reply bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java]: The common part of requesting subscription url is now combined into a common method.

File: bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java [hunk 2]
```diff
@@ -297,7 +283,34 @@ public void updateRequest() throws IOException {
...
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Unexpected subscription result from the server");
+            } else {
+                logger.debug("Reconnecting Subscription to: {}", subscriptionURL);
+                open();
+            }
+        }
+    }
+
+    public void getSubscriptionUrl() throws IOException {
```
Comment (jlaur) [inline bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java:299]: It looks like this method could be private?

Comment (kjoglum) [reply bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java]: Method made private.

**Review by jlaur — APPROVED** [review]
LGTM

**Pull request merged**